test/jdk/jdk/nio/zipfs/TestPosix.java

```diff
@@ -733,6 +733,33 @@ public void setPermissionsShouldPreserveRemainingBits() throws IOException {
         });
     }
 
+    /**
+     * Verify that calling Files.setPosixFilePermissions on an MS-DOS entry
+     * results in only the expected permission bits being set
+     *
+     * @throws IOException if an unexpected IOException occurs
+     */
+    @Test
+    public void setPermissionsShouldConvertToUnix() throws IOException {
+        // The default environment creates MS-DOS entries, with zero 'external file attributes'
+        createEmptyZipFile(ZIP_FILE, ENV_DEFAULT);
+        try (FileSystem fs = FileSystems.newFileSystem(ZIP_FILE, ENV_DEFAULT)) {
+            Path path = fs.getPath("hello.txt");
+            Files.createFile(path);
+        }
+
+        // Sanity check that all 'external file attributes' bits are zero
+        verifyExternalFileAttribute(Files.readAllBytes(ZIP_FILE), "0");
+
+        // Convert to a UNIX entry by calling Files.setPosixFilePermissions
+        try (FileSystem fs = FileSystems.newFileSystem(ZIP_FILE, ENV_POSIX)) {
+            Path path = fs.getPath("hello.txt");
+            Files.setPosixFilePermissions(path, EnumSet.of(OWNER_READ));
```

Comment:
It might be helpful to show the before/after output of the CEN fields here just for extra clarity

Reply:
> It might be helpful to show the before/after output of the CEN fields here just for extra clarity

Thanks! Can you take a look at 4bb77e1 which shows the `zipdetails` output for relevant fields before/after calling `setPosixFilePermissions`.

Reply:
thank you Looks good